test/e2e/framework/ssh/ssh.go

```diff
@@ -285,3 +303,61 @@ func IssueSSHCommand(cmd, provider string, node *v1.Node) error {
 	}
 	return nil
 }
+
+// nodeAddresses returns the first address of the given type of each node.
+func nodeAddresses(nodelist *v1.NodeList, addrType v1.NodeAddressType) []string {
```

Comment:
I'll let this pass, but I'd bet we have a bunch of overlap with other functions.

Reply:
It is 100% copied so I'm positive we do. It was part of the `framework` though where we are actively trying to remove code from. My understanding was that we expected a number of small methods like this to be duplicated across the codebase in order to support the decoupling.